DPL: do not add extra DataHeader to EoS

ktf opened the pull request:
DPL: do not add extra DataHeader to EoS

Historically we had to have a DataHeader for EoS in any case.

This is not needed anymore and it might explain some weird behavior we
see online.

ktf commented:
Apparently since ever the OutputProxy was sending data headers together with the SourceInfoHeader. As far as i can tell, this is the only place where we actually do, and from @davidrohr description of the issue with output proxy and calibration, I wonder if this is not causing it.

ktf commented:
Notice that in principle one should have received a warning about the unexpected data in input...

davidrohr commented:
For reference, tested this, and we saw some issues with "Data not following O2 data model" on staging, so should not be merged yet.

ktf committed:
DPL: do not add extra DataHeader to EoS

Historically we had to have a DataHeader for EoS in any case.

This is not needed anymore and it might explain some weird behavior we
see online.

ktf commented:
@davidrohr As far as I can tell, Data Distribution does not use DPL/EOS anymore. If I understand correctly, the message comes from the fact the proxy checks for correctness on send, so the message only appears when it's DPL injecting the EoS, rather than DD. The updated commit should fix the issue.

ktf commented:
In the end this is not actually correct, so we are closing this.
